mozdownload/scraper.py

```diff
@@ -819,11 +819,16 @@ def binary_regex(self):
 
     def build_filename(self, binary):
         """Return the proposed filename with extension for the binary."""
-        return '%(TIMESTAMP)s%(BRANCH)s%(DEBUG)s-%(NAME)s' % {
-            'TIMESTAMP': self.timestamp + '-' if self.timestamp else '',
+        if not self.timestamp and self.date:
+            date = self.date.strftime('%Y-%m-%d-')
```

Comment:
I think that we should try to always use the timestamp. Shouldn't we be able to here to use it given that we have the subfolder name of the specific build on that day? Please also keep in mind that we can have several tinderbox builds per day, so only the date would cause duplicate filenames.  For the latter we should get another test added.